sentry-ruby/spec/sentry/event_spec.rb

```diff
@@ -141,6 +141,12 @@
         expect(event.to_hash[:user][:ip_address]).to eq("2.2.2.2")
       end
 
+      it "doesn't overwrite already set ip address" do
+        Sentry.set_user({ ip_address: "3.3.3.3" })
+        Sentry.get_current_scope.apply_to_event(event)
+        expect(event.to_hash[:user][:ip_address]).to eq("3.3.3.3")
```

Comment:
@sl0thentr0py note for the future - this should be `event.to_h`, because `to_hash` is a special coercion method that's used implicitly in some places in Ruby and may cause weird issues. For example `**` uses it, see:

```ruby
(ruby) {foo: "bar", **event}
{:foo=>"bar",
 :event_id=>"266e3389d9b64f57b48da6f3d0969a93",
 :level=>:error,
 :timestamp=>"2024-07-22T16:22:13Z",
 :environment=>"development",
 :server_name=>"44b9b235e7c3",
 :modules=>
  {"rake"=>"12.3.3",
 # ...
```

It should only be used when the object that implements it is actually a hash-like object, and events are not like that.

Reply:
Thanks! For context, I took it from the [old SDK](https://github.com/getsentry/sentry-ruby/blob/36866c54ea67933c36a5d8ce2b52dae972e3edd1/sentry-raven/lib/raven/event.rb#L135-L148) when implementing `sentry-ruby`. So we probably can make the switch rather freely. I'll put up a PR to see how it goes.